infrastructure/cdn-in-a-box/docker-compose.yml

```diff
@@ -209,48 +209,36 @@ services:
       - variables.env
     hostname: trafficvault
 
-  # Apache Traffic Server (ATS) caches defined here
-  # base image from which all other caches inherit (builds and installs ATS+plugins)
-  tccache:
-    build:
-      context: ../..
-      dockerfile: infrastructure/cdn-in-a-box/cache/Dockerfile
-    image: tccache:latest
-
   # edge cache
   edge:
     privileged: True
     build:
-      context: .
-      dockerfile: edge/Dockerfile
-    depends_on:
-      - enroller
-      - tccache
+      context: ../..
+      dockerfile: infrastructure/cdn-in-a-box/cache/Dockerfile
     domainname: infra.ciab.test
     env_file:
       - variables.env
     hostname: edge
     volumes:
       - ./dns/set-dns.sh:/usr/local/sbin/set-dns.sh
       - ./dns/insert-self-into-dns.sh:/usr/local/sbin/insert-self-into-dns.sh
+      - ./edge/run.sh:/opt/init.d/99-run.sh
```

Comment:
Why put this in `/opt/init.d` instead of just `/`? It seems like you're setting this up to run a bunch of scripts in a row, but there's only the one. If this was just added to container so that the `CMD` called it directly you could get rid of the whole `cache/run.sh` file you made to shell out calls to the real run script.

Reply:
I want to let this container could be launched without any other injection.
That why I add another `run.sh` to drive another injected startup script.
If we add `CMD /run.sh` but not add a real `run.sh` in the container, we can't run `docker run cdn-in-a-box_edge` directly.
How do you think?

Reply:
Well, I think normally the way we'd do something like that is with a build argument using `ARG`

Reply:
It's different. If you build different images through different ARG, that means the built images will be different. I mean the image id will be different.
What I want to do is to keep to use the only image. The role it plays is dependent on runtime parameters (file(s) be put in `/opt/init.d`)

Reply:
Yeah, that's true, they would be different images if we used `ARG`. I guess I just balk at the idea of running a bunch of things in one container, and this sort of seems to be moving in that direction. I think a volume is fine implementation, I just hope this doesn't turn into an orchestration system for a highly complex container.

Reply:
I don't quite understand what you're saying `a volume is fine implementation`. Could you give me a hint?

Reply:
Nothing, it's fine